### PR TITLE
[FEATURE] Community roadmap added to the community dropdown in Nav

### LINKED ISSUE #3292: [Feature Request] Project Roadmap -> Navbar
### Is your feature request related to a problem? Please describe

Add project roadmap back into the navbar. 
https://github.com/orgs/opensearch-project/projects/220

### Describe the solution you'd like to see

Maybe into documentation? 

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by kaimmej**
### Description
Added a project roadmap link to the community section of the navbar
 
### Issues Resolved
#3292 

### Check List
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the BSD-3-Clause License.

### CODE REVIEW

File: _data/header_navmenu_datafile.json [hunk 1]
```diff
@@ -328,6 +328,13 @@
                             "URL": "https://github.com/opensearch-project",
                             "ClassValue" : "drawer-content-subheader-link",
                             "Icon": "OpenSearch-Github-condensedImage"
+                        },
+                        {
+                            "Name": "Project Roadmap",
+                            "SubText": "What is next for the OpenSearch project?",
+                            "URL": "https://github.com/orgs/opensearch-project/projects/220",
```
Comment (peterzhuamazon) [inline _data/header_navmenu_datafile.json:335]: Hi @kaimmej the new roadmap is this: https://github.com/orgs/opensearch-project/projects/206
```suggestion
                            "URL": "https://github.com/orgs/opensearch-project/projects/206",
```

cc: @Pallavi-AWS @getsaurabh02 

Thanks!

Comment (getsaurabh02) [reply _data/header_navmenu_datafile.json:335]: We are moving towards new dashboard: https://github.com/orgs/opensearch-project/projects/206

Comment (kaimmej) [reply _data/header_navmenu_datafile.json]: Thank you for the feedback! 
### Updates:
- I have updated the project roadmap link to point to the correct project
- Additionally, I added a card that distinguishes it from the other navbar items. 
- Roadmap moved to the docs dropdown. 

### Image
image

Comment (krisfreedain) [reply _data/header_navmenu_datafile.json]: this looks great @kaimmej

**Review by krisfreedain — APPROVED** [review]
thanks @kaimmej

**Pull request merged**